### DISCUSSION TITLE
SwiftSyntax parser library isn't compatible

### DISCUSSION THREAD

**Discussion by eshi13** [uber/piranha#654] [Q&A]
I was trying to build the code using `swift build && swift package generate-xcodeproj` but it throws an error
```
error: no such module '_InternalSwiftSyntaxParser'
@_implementationOnly import _InternalSwiftSyntaxParser
```
If I change the version for swift syntax from 0.50500.0 to 0.50600.0, the error changes to 
error: cannot find 'SyntaxParser' in scope
        let parsed = try SyntaxParser.parse(sourceFileURL)
                         ^~~~~~~~~~~~
and in version 510.0.1, there are multiple errors, can anyone help with this?

**Comment by ketkarameya** [comment]
Hey! Thanks for trying out Piranha. I was wondering could you try the polyglot-piranha variant?  We do not have have any maintainer for the legacy variants

**Reply by eshi13** [reply]
Hi @ketkarameya , thanks for replying. Is there any jupyter notebook or any example demo that can help me in onboarding. I am little confused with rules and scope

**Reply by ketkarameya** [reply]
have you tried to play around with the demos https://github.com/uber/piranha/tree/master/demo

**Reply by ketkarameya** [reply]
@eshi13  Are u unblocked?

**Reply by eshi13** [reply]
yeah @ketkarameya , I made some progress. Thank you

**Reply by eshi13** [reply]
Just one small doubt. I wrote this rule to replace ccmContainer.stale_flag name instead of stale_flag.isEnabled. But it doesn't seem to work, does this seem wrong to you?
```
[[rules]]
name = "test_rule_replace_true_placeholder"
query = """(
(navigation_expression
        target: (navigation_expression
            target: (simple_identifier)
            suffix: (navigation_suffix
                suffix: (simple_identifier) @access_identifier))
        suffix: (navigation_suffix
            suffix: (simple_identifier) @param)) @parameter_access
(#eq? @access_identifier "ccmContainer")
(#eq? @param "@stale_flag")
)"""
replace_node = "parameter_access"
replace = "@treated"
holes = ["stale_flag", "treated"]
groups = ["replace_expression_with_boolean_literal"]
```
